### PR TITLE
PR template

### PR DESCRIPTION
**Pull request by ArnisLielturks**
Template for all submitted PR's which contains a checklist for  things to do in the PR before it could be merged. It would also work as a reminder for the author and reviewers what should be taken into account when reviewing it.

This PR is open for discussion what things should be added/removed from the list and if we need such functionality at all.

### CODE REVIEW

**Comment by 1vanK** [comment]
I highly recommend not oblige contributors to add bindings if they don't want to. It stops many. The contributors is only obliged to correct the bindings when changing existed functionality, so that after his PR the project still to be compiled. Those interested in the bindings will always be able to add them on their own (this has happened more than once).

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+- [ ] Bug
+- [ ] Feature
+- [ ] Improvement
+- [ ] Documentation
+
+### To-do
+
+- [ ] Code is formatted according to the guidlines [Coding conventions](https://urho3d.github.io/documentation/HEAD/_coding_conventions.html)
+- [ ] Angel Script bindings include my changes (if there are any) [AS bindings](https://github.com/urho3d/Urho3D/tree/master/Source/Urho3D/LuaScript)
```
Comment (1vanK) [inline .github/pull_request_template.md]: AS links to Lua

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+- [ ] LUA bindings include my my changes (if there are any) [LUA bindings](https://github.com/urho3d/Urho3D/tree/master/Source/Urho3D/AngelScript)
```
Comment (1vanK) [inline .github/pull_request_template.md]: Lua links to AS

Comment (1vanK) [reply .github/pull_request_template.md]: "include my my"

**Comment by ArnisLielturks** [comment]
@1vanK I agree that we should not force the authors to update the bindings. Are you proposing that we remove the bindings block from the template or just state that it's not obligatory? The sole purpose of this PR is to make the review process faster and a bit cleaner for all the future PR's.

**Comment by weitjong** [comment]
May be add a link to https://urho3d.github.io/documentation/HEAD/_contribution_checklist.html

Until the auto-binding tool is available, the requestor may get other help to get the binding done. The merge can wait until then.

**Comment by ArnisLielturks** [comment]
@Modanung had a great idea that we could add the issue templates too.

They could also contain links to the forum and github's wiki page to avoid duplicate issues + automatic tagging.

**Comment by weitjong** [comment]
BTW, you can add "[ci skip]" anywhere in the commit message to skip the CI build for non code change commits. Saving Gaia.

**Comment by 1vanK** [comment]
Currently we use Doxygen for generation of docs. Doxygen fully parses C++ sources, so has full information about classes. Is it possible to use it information to generate bindings?

http://www.doxygen.nl/manual/features.html
> Generates structured XML output for parsed sources, which can be used by external tools.

**Comment by Modanung** [comment]
@1vanK I took your suggestion to the [forums](https://discourse.urho3d.io/t/script-binding-automation/5921/42).

**Pull request closed without merging**